runtime/astar/src/precompiles.rs

```diff
@@ -45,6 +48,30 @@ use xcm::latest::prelude::MultiLocation;
 /// to Erc20AssetsPrecompileSet
 pub const ASSET_PRECOMPILE_ADDRESS_PREFIX: &[u8] = &[255u8; 4];
 
+/// Filter that only allows whitelisted runtime call to pass through dispatch precompile
+#[derive(Default)]
+pub struct DispatchPrecompileFilter;
```

Comment:
Good naming documents itself. I would recommend to
- Rename it as `WhitelistedCalls`
- Use `frame_support::traits::Contains` trait

The usage will be like `WhitelistedCalls::contains(&call)`, self-explanatory.